template.tpl

```diff
@@ -2792,6 +2848,52 @@ const processStackAdaptEvent = () => {
   }
 };
 
+const processPinterestAdsEvent = () => {
+  const pinterestSDKKey = "pinterest-ads";
+  let options = generateOptions(pinterestSDKKey);
+
+  if (data.commonEventName) {
+    const props = parseSimpleTable(data.commonEventProperties || []);
+
+    track(data.commonEventName, props, options);
+
+    data.gtmOnSuccess();
+  } else {
+    log("ERROR: Freshpaint Pinterest Ads GTM Template missing eventName");
+    data.gtmOnFailure();
+  }
+};
+
+const processRedditAdsEvent = () => {
+  const redditSDKKey = "reddit-ads";
+  let options = generateOptions(redditSDKKey);
+
+  if (data.commonEventName) {
+    // TODO: Add bool parse support to either backend for string or here to cast to bool, for: test_mode, opt_out props if present
+    const props = parseSimpleTableAndParseNumericAndJSONValues(data.commonEventPropertiesJSONValue || []);
+
+    let instanceNameToUse;
+    if (data.redditAdsInstanceName) {
+      instanceNameToUse = data.redditAdsInstanceName.trim();
+    }
+    if (instanceNameToUse) {
+      options = generateOptionsFromInstances(redditSDKKey, instanceNameToUse, false);
```

Comment:
I want to talk about the `false` here which looks like it prevents the same event from going to multiple configs at the same time, but that's not a blocker.